### ISSUE TITLE
The block switcher 'Transform' menu should not contain extraneous content

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#67579]
### Description

As already pointed out in other issues and PRs, ARIA menus should only contain menuitems (and their varints menuitemradio / menuitemcheckbox), groups and separators.

The 'Transform to' menu in the block toolbar is an ARIA menu, it does have a `role="menu"` and does provide the expected semantics and intercion of the ARIA menu design pattern.

However, there are at least two cases where this menu mayu contain extraneous content that breaks the ARIA design pattern:

- When there are no transforms available.
- When a block attribute is 'connected' to a source via the Bindings API.

Screenshot:

![Image](https://github.com/user-attachments/assets/12103a57-6685-43f8-9bd4-2845fbba66bd)

Code reference:

https://github.com/WordPress/gutenberg/blob/88a3cef341fac21d74a08b0d94e0b0bc45c0fbd6/packages/block-editor/src/components/block-switcher/index.js#L122-L128

and 

https://github.com/WordPress/gutenberg/blob/88a3cef341fac21d74a08b0d94e0b0bc45c0fbd6/packages/block-editor/src/components/block-switcher/index.js#L178-L184

An ARIA menu is not a place for additional information or descriptions. These informational text should be either removed from teh menu or places somewhere else.

1. It's not clear to me why there's the need to render the menu when there are no transforms available. Instead of rendering the menu with a paragraph with text `No transforms.`, the block switcher toggle button should be aria-disabled and the menu not rendered in the first place.
2. The 'connected' text is just a `<span>` element within an ARIA menu group. Thi sbreaks hte ARIA pattern and should be moved elsewhere.

### Step-by-step reproduction instructions

- Go to the Site editor.
- Select a template part e.g. 'Header'.
- Use the keyboard to navigate to the Header block toolbat (Shift+Tab).
- Observe that you can open the block switcher menu by pressing Enter or Spacebar on the first button in the block toolbar.
- However, observe that there is no way to move focus or navigate by arrow keys to the content of the menu so that the 'No transforms' information is not perceivable by screen reader users.

Optionally:
- Connect a block attribute to a source via the Bindings API. Instructions on how to use the Bindings API are a little too long to be provided here. This step is not strictly necessary because it's basically the same issue of the 'no transforms' text described above.
- Select the block.
- Use the keyboard to navigate to the block toolbar (Shift+Tab).
- Open the block switcher menu.
- Observe there is no way to navigate to the 'This block is connected.' text when tabbing or using the arrow keys.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by cbravobernal** [comment]
> The 'connected' text is just a <span> element within an ARIA menu group. This breaks the ARIA pattern and should be moved elsewhere.

Any idea of where could we move it?

**Comment by afercia** [comment]
> Any idea of where could we move it?

I started working on a PR where I'm trying to move it to the settings panel, in the card at the top. So far, I just moved the functionality and the text message and haven't used any styling for now. It may benefit from some icon and a little design touch. Rough screenshot of what I have so far:

![Image](https://github.com/user-attachments/assets/00f4eac3-8250-4e6e-bdf5-dd765de77f1e)

**Comment by afercia** [comment]
Noting that the 'No transforms.` text is shown in the block switcher also in other cases. For example, a list item block in a List block doesn't have transforms or styles available:

![Image](https://github.com/user-attachments/assets/dd165c8e-0bfd-4a7e-a2e0-3676ff886f1c)

**Comment by richtabor** [comment]
Left a review on the pr: 

> This is not an appropriate design for this change. It cannot scale well, particularly when there is any other elements rendered in the block card. Nor is it particularly helpful having this buried in the block transforms menu.

> I propose just removing it as originally intended, but not adding it to the block card.

> There are existing patterns we could potentially explore as follow-ups, such as following the block name convention ("Main (Group)") with "Connected" or rendering the connected icon (or the "Connected" label perhaps) in a badge to the right of the block name, as seen in List view for anchor links. I lean towards the latter, if this is significant enough to render.

**Comment by afercia** [comment]
Quick comparison of potential approaches for the 'connected info' in the post card:

![Image](https://github.com/user-attachments/assets/a079ad71-62a7-4126-a2ce-b03fa27b23c9)

I'm not sur ethe badge inline with the block name would be ideal. Blocks may have very longer names and translations may provide longer strings so that this inline layout may be problematic.

Thinking at the overall picture, I'm wondering whether it's worth to consider a more generic way to show information related to some block 'properties'. It could be 'connected' for bound blocks or 'synced' for patterns or any otehr relevant block property worth highlighting in the post card. Would a series of badges (sort of 'tags') after the block name and description be a good potential approach?

**Comment by carolinan** [comment]
Can we split this into separate smaller issues? One for the transform and one for the block bindings? I am finding it disorientating.
